### PR TITLE
[Course] F2: Courses Index page with tests and stories

### LINKED ISSUE #34: [Course] F2: Courses Index Page with tests and stories
- [ ] When clicked, courses button on navbar takes you to the Courses Index page
- [ ] On the courses index page, there is a button to take you to a working create page, or a placeholder page.
- [ ] The courses index page has a table of all the courses 
- [ ] The courses table has a delete button that allows you to delete a course
- [ ] The courses table has an edit button that either takes you to a working edit page, or a placeholder page.

### PR DESCRIPTION
**Pull request by justinchung712**
- Courses index page at /courses
- Normal table for regular user
- Buttons to create, edit, and delete courses appear for admins and instructors
- Storybook at: https://ucsb-cs156-f23.github.io/proj-organic-f23-6pm-2/prs/68/storybook/?path=/story/pages-courses-coursesindexpage--empty (three items will not show in table in stories unless ran locally because of login requirement)
- Dokku deployment at: https://oj-dev.dokku-06.cs.ucsb.edu (updated)

Closes #34

### CODE REVIEW

**Review by whuang602 — APPROVED** [review]
Beautiful

File: frontend/src/tests/pages/Courses/CoursesIndexPage.test.js [hunk 1]
```diff
@@ -35,10 +102,102 @@ describe("CoursesIndexPage tests", () => {
...
+        // assert that the Create button is not present when user isn't an admin or instructor
+        expect(screen.queryByText(/Create Course/)).not.toBeInTheDocument();
+    });
+
+    test("renders empty table when backend unavailable, user only", async () => {
+        setupUserOnly();
+        const queryClient = new QueryClient();
+        axiosMock.onGet("/api/courses/all").timeout();
+
+        // const restoreConsole = mockConsole();
```
Comment (pconrad) [inline frontend/src/tests/pages/Courses/CoursesIndexPage.test.js:118]: commented out code stilll

File: frontend/src/tests/pages/Courses/CoursesIndexPage.test.js [hunk 2]
```diff
@@ -35,10 +102,102 @@ describe("CoursesIndexPage tests", () => {
...
+                    <CoursesIndexPage />
+                </MemoryRouter>
+            </QueryClientProvider>
+        );
+
+        await waitFor(() => { expect(axiosMock.history.get.length).toBeGreaterThanOrEqual(1); });
+        
+        // const errorMessage = console.error.mock.calls[0][0];
+        // expect(errorMessage).toMatch("Error communicating with backend via GET on /api/courses/all");
+        // restoreConsole();
```
Comment (pconrad) [inline frontend/src/tests/pages/Courses/CoursesIndexPage.test.js:132]: Still commented out code.

**Review by pconrad — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**